pinot-core/src/main/java/org/apache/pinot/core/common/BlockValSet.java

```diff
@@ -84,6 +85,13 @@
    */
   double[] getDoubleValuesSV();
 
+  /**
+   * Returns the BigDecimal[] values for a single-valued column.
+   *
+   * @return Array of BigDecimal[] values
+   */
+  BigDecimal[] getBigDecimalValuesSV();
```

Comment:
This API is not required as we don't want to make `BigDecimal` as a storage type

Reply:
@Jackie-Jiang Added it for convenience to avoid calling two APIs: `getBytesValuesSV` and `ArrayCopyUtils.copy` everywhere.

Will remove and move the logic higher up to different transforms.

Reply:
I don't follow. We should just use `getBytesValuesSV` for `BigDecimal` types